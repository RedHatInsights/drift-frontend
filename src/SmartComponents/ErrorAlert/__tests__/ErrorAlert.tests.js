import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';

import ConnectedErrorAlert, { ErrorAlert } from '../ErrorAlert';

describe('ErrorAlert', () => {
    let props;

    beforeEach(() => {
        props = {
            error: { status: 404, detail: 'This is an error' },
            onClose: jest.fn(),
            addNotification: jest.fn()
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(
            <ErrorAlert { ...props } />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    /*eslint-disable no-unused-vars*/
    it('should call onClose', () => {
        const wrapper = mount(
            <ErrorAlert { ...props } />
        );

        expect(props.onClose).toHaveBeenCalled();
    });

    it('should call onClose with tableId', () => {
        props.tableId = 'COMPARISON';
        const wrapper = mount(
            <ErrorAlert { ...props } />
        );

        expect(props.onClose).toHaveBeenCalledWith('COMPARISON');
    });
    /*eslint-enable no-unused-vars*/
});

describe('ConnectedErrorAlert', () => {
    let initialState;
    let props;
    let mockStore;

    beforeEach(() => {
        mockStore = configureStore();

        initialState = {
            addNotification: jest.fn()
        };

        props = {
            error: { status: 404, detail: 'This is an error' },
            onClose: jest.fn(),
            addNotification: jest.fn()
        };
    });

    it.skip('should render correctly', () => {
        const store = mockStore(initialState);
        const wrapper = mount(
            <MemoryRouter keyLength={ 0 }>
                <Provider store={ store }>
                    <ConnectedErrorAlert { ...props } />
                </Provider>
            </MemoryRouter>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
