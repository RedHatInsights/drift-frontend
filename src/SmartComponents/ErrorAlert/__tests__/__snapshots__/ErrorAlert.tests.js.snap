// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorAlert should render correctly 1`] = `<Fragment />`;
